book/src/template_expansion.md

```diff
@@ -0,0 +1,387 @@
+# Template Expansion
+
+This chapter will explain how the different parts of the templates are
+translated into Rust code.
+
+## Basic explanations
+
+Whwn you add `#[derive(Template)]` and `#[template(...)]` on your type, the
+`Template` derive proc-macro will then generate an implementation of the
+`askama::Template` trait which will be a Rust version of the template.
+
+It will also implement the `std::fmt::Display` trait on your type which will
+internally call the `askama::Template` trait.
+
+Let's take a small example:
+
+```rust
+#[derive(Template)]
+#[template(source = "{% set x = 12 %}", ext = "html")]
+struct Mine;
+```
+
+will generate:
+
+```rust
+impl ::askama::Template for YourType {
+    fn render_into(
+        &self,
+        writer: &mut (impl ::std::fmt::Write + ?Sized),
+    ) -> ::askama::Result<()> {
+        let x = 12;
+        ::askama::Result::Ok(())
+    }
+    // I skipped some internal `askama` impl items to keep this as simple
```

Comment:
I would probably retain these extra items at least once to give people a better sense of what's going on. It's not all that much, right?

Reply:
It's only some associated constants, but I don't think they bring much value. I can put them back though if you think they're useful to see.

Reply:
Yes, I think it's good to give the full output once for the most minimal template.